File: pharmacy-api/shared/setup/broker_connect.go
package setup

import (
	"fmt"
	"log"
	"os"

	shared_models "pharmacy-api/shared/models"

	amqp "github.com/rabbitmq/amqp091-go"
)

func ConnectBroker() *shared_models.Broker {
	connString := fmt.Sprintf("amqp://%s:%s@%s:5672/",
		os.Getenv("RABBITMQ_USER"),
		os.Getenv("RABBITMQ_PASSWORD"),
		os.Getenv("RABBITMQ_HOST"),
	)
	conn, err := amqp.Dial(connString)
	if err != nil {
		log.Fatal("failed to connect to broker:", err)
	}

	ch, err := conn.Channel()
	if err != nil {
		conn.Close()
		log.Fatal("failed to open channel:", err)
	}

	return &shared_models.Broker{
		Connection: conn,
		Channel:    ch,
	}
}

File: pharmacy-api/src/shared/setup/broker_connect.ts
import * as amqp from 'amqplib';
import { Broker } from '@pharmacy/src/shared/models/models';

export async function connectBroker(): Promise<Broker> {
  const connString = `amqp://${process.env.RABBITMQ_USER}:${process.env.RABBITMQ_PASSWORD}@${process.env.RABBITMQ_HOST}:5672/`;
  
  try {
    const connection = await amqp.connect(connString);
    console.log('Connected to RabbitMQ successfully');
    
    const channel = await connection.createChannel();
    console.log('RabbitMQ channel created successfully');
    
    return {
      connection: connection,
      channel: channel,
    };
  } catch (error) {
    console.error('Failed to connect to broker:', error);
    throw error;
  }
}

export async function closeBroker(broker: Broker): Promise<void> {
  try {
    if (broker.channel) {
      await broker.channel.close();
      console.log('RabbitMQ channel closed');
    }
    if (broker.connection) {
      await broker.connection.close();
      console.log('RabbitMQ connection closed');
    }
  } catch (error) {
    console.error('Error while closing broker connection:', error);
    throw error;
  }
}